webrender/src/frame_builder.rs

```diff
@@ -46,12 +46,21 @@ fn make_polygon(
     stacking_context: &StackingContext,
     node: &ClipScrollNode,
     anchor: usize,
+    device_pixel_ratio: f32,
 ) -> Polygon<f32, WorldPixel> {
-    //TODO: only work with `isolated_items_bounds.size` worth of space
     // This can be achieved by moving the `origin` shift
     // from the primitive local coordinates into the layer transformation.
     // Which in turn needs it to be a render task property obeyed by all primitives
     // upon rendering, possibly not limited to `write_*_vertex` implementations.
+    if stacking_context.isolated_items_bounds.is_empty() {
+        // isolated_items_bounds is empty means there is no item for this sc.
+        // in this case, we use screen_bounds instead.
+        let rect = stacking_context.screen_bounds.to_f32() / device_pixel_ratio;
+        let bounds = LayerRect::new(LayerPoint::new(rect.origin.x, rect.origin.y),
+                                    LayerSize::new(rect.size.width, rect.size.height));
+        return Polygon::from_transformed_rect(bounds, LayerToWorldTransform::identity(), anchor);
```

Comment:
hmm, I'm not seeing the point here. If a stacking context has no items to isolate, why would we even want it to participate in the plane splitting (and hence, generate a polygon here)?

Reply:
In the reftest we have following display list.
```
sc1 (preserve3d)
  sc2 (preserve3d)
    rect1
  sc3 (flat)
    sc4 (preserve3d)
      sc5 (preserve3d)
        rect2
      sc6 (preserve3d)
        rect3
```

We'll create plane splitting for rect 2 and rect 3, then render it in sc3. Next, we'll create plane splitting for sc3 and render it on the screen. Thus, we have a sc with no item bound but need a plane splitting. Does it makes sense?

Reply:
I believe the `isolated_items_bounds` for SC3 would then need to include the bounds of SC4. Given that SC4 is transformed, its bounds would need to be projected onto SC3 (transformed).